libnetwork/portmapper/proxy_linux.go

```diff
@@ -61,6 +66,17 @@ func (p *proxyCommand) Start() error {
 	}
 	w.Close()
 
+	go func() {
+		p.waitOnce.Do(func() {
+			p.cmdWaitError = p.cmd.Wait()
```

Comment:
This goroutine should be the only place we call `Wait()`.
From here we can signal others that wait is completed (see [sync.Cond](https://pkg.go.dev/sync#Cond)).

I do not think we should bother with the `waitOnce` and instead just call it once.
I'd also be really surprised if the race detector doesn't complain about read/write on `cmdWaitError` (even if it technically shouldn't be a problem).

Reply:
Yeah, was thinking the same; we'd still have to capture the error though (probably); see https://github.com/moby/moby/pull/43598#discussion_r872976255

Reply:
Indeed.
With a condvar you can do something like:

```go
p.waitMu.Lock()
defer p.waitMu.Unlock()
while !p.done() { p.waitCond.Wait() }
return p.waitErr
```

Reply:
@cpuguy83